src/test/java/org/example/UserTest.java

```diff
@@ -26,133 +20,109 @@
-        String response = client.execute(request, new BasicHttpClientResponseHandler());
-        List<ValidationMessage> errors = validatePayload(response, "getUserTemplate");
-        Assert.assertTrue(errors.isEmpty());
+    public void testGetUser() throws IOException, ParseException {
+        CloseableHttpResponse response = httpService.sendGetRequest(testDataFactory.getId());
+        int responseCode = response.getCode();
+        String responseBody = EntityUtils.toString(response.getEntity());
+        Assert.assertEquals(responseCode, HTTP_OK, "Wrong response code. Expected: " + HTTP_OK + " Received: " + responseCode);
+        List<ValidationMessage> errors = jsonValidator.validatePayload(responseBody, "getUserTemplate");
+        Assert.assertTrue(errors.isEmpty(), errors.toString());
     }
 
     @Test
-    public void getNonExistentUser() throws IOException {
-        HttpUriRequest request = new HttpGet("https://gorest.co.in/public/v2/users/1");
-        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
-        Assert.assertEquals(httpResponse.getCode(), 404);
+    public void testGetNonExistentUser() throws IOException {
+        CloseableHttpResponse response = httpService.sendGetRequest(NON_EXISTENT_ID);
+        int responseCode = response.getCode();
+        Assert.assertEquals(responseCode, HTTP_NOTFOUND, "Wrong response code. Expected: " + HTTP_NOTFOUND + " Received: " + responseCode);
     }
 
     @Test
-    void postUser() throws IOException {
-        CloseableHttpClient client = HttpClients.createDefault();
-        HttpPost httpPost = new HttpPost(String.format("https://gorest.co.in/public/v2/users"));
-        List<NameValuePair> params = new ArrayList<NameValuePair>(2);
-        params.add(new BasicNameValuePair("name", "John Doe"));
-        params.add(new BasicNameValuePair("email", String.format("%s@gmail.com", RandomStringUtils.random(10, true, true))));
-        params.add(new BasicNameValuePair("gender", "male"));
-        params.add(new BasicNameValuePair("status", "active"));
-        httpPost.setEntity(new UrlEncodedFormEntity(params));
-        Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
-        httpPost.setHeader(header);
-        String response = client.execute(httpPost, new BasicHttpClientResponseHandler());
-        List<ValidationMessage> errors = validatePayload(response, "postUserTemplate");
-        Assert.assertTrue(errors.isEmpty());
+    void testCreateUser() throws IOException, ParseException {
+        CloseableHttpResponse response = httpService.sendPostRequest(testDataFactory.generateTestUser());
+        int responseCode = response.getCode();
+        String responseBody = EntityUtils.toString(response.getEntity());
+        Assert.assertEquals(responseCode, HTTP_CREATED, "Wrong response code. Expected: " + HTTP_CREATED + " Received: " + responseCode);
```

Comment:
use HttpStatus interface with statuses from package org.apache.http;

Reply:
I think you ca add a validation step to check that the returned user data matches the expected input